serviceconnector/src/createServiceConnector/createLinker.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { KnownClientType } from "@azure/arm-servicelinker";
+import { AzExtTreeItem, AzureWizard, AzureWizardExecuteStep, AzureWizardPromptStep, ExecuteActivityContext, IActionContext, createSubscriptionContext, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import * as vscode from 'vscode';
+import { AuthenticationListStep } from "./AuthenticationListStep";
+import { ClientListStep } from "./ClientListStep";
+import { ICreateLinkerContext } from "./ICreateLinkerContext";
+import { LinkerCreateStep } from "./LinkerCreateStep";
+import { LinkerNameStep } from "./LinkerNameStep";
+import { TargetServiceListStep } from "./TargetServiceListStep";
+
+export interface LinkerItem {
+    subscription: AzureSubscription,
+    id: string,
+}
+
+export async function createLinker(context: IActionContext & ExecuteActivityContext, item: LinkerItem | AzExtTreeItem, preSteps: AzureWizardPromptStep<ICreateLinkerContext>[] = [], runtime?: KnownClientType[]): Promise<void> {
```

Comment:
`preSteps` isn't optional, but it's not clear to me (just from looking at this method) what the preSteps need to accomplish in order to work properly. Are there specific properties on `ICreateLinkerContext` the preSteps should define?

If so, a comment would go a long way here.

Reply:
`preSteps` is referring to steps that are specific to the client extension that should to be set before any of the general service connector steps. For example in container apps we need to set the container. I had originally had this as an optional parameter but ran into issues with that since I was using nonNullValue which caused an error when testing with app service since there are no pre steps for that service. For more context here is the comment Nathan left: 

>You shouldn't assume that this is a nonNullValue. It'll cause problems in the event no prompt steps are passed through.

>You should could probably just make preSteps default to an empty array:

>`export async function createLinker(context: IActionContext & ExecuteActivityContext, item: LinkerItem | >AzExtTreeItem, preSteps: AzureWizardPromptStep<ICreateLinkerContext>[] = []): Promise<void>`

Reply:
Oh I'm sorry, I totally missed that it was defaulting to an empty array. That makes more sense.